### PR TITLE
Cancel change request

### PR DESCRIPTION
**Pull request by andrewfraser73**
contains code for cancelling change request in part or in full (making live will be the same)
-  if you don't click any bom or pcas changes it will cancel everything
- if you click individual bom or pcas changes only they will be cancelled.  if you cancel everything on the change request it will cancelled

also a couple of UX changes
- bom/pcas changes tabs now show number of changes or disabled if none
- change status now colour coded as per jacki

### CODE REVIEW

File: src/Resources/BomChangeResource.cs [hunk 1]
```diff
@@ -1,5 +1,7 @@
 ﻿namespace Linn.Purchasing.Resources
 {
+    using System.Runtime.InteropServices;
```
Comment (lewisrenfrew) [inline src/Resources/BomChangeResource.cs:3]: probably sneaked in

File: src/Service.Host/client/src/components/ChangeRequests/ChangeRequest.js [hunk 2]
```diff
@@ -35,20 +41,82 @@ function ChangeRequest() {
...
+        }
+
+        if (
+            changereq.pcasChanges.filter(c => c.changeState === 'LIVE').length >
+            origreq.pcasChanges.filter(c => c.changeState === 'LIVE').length
+        ) {
+            return true;
+        }
+
+        return false;
```
Comment (lewisrenfrew) [inline src/Service.Host/client/src/components/ChangeRequests/ChangeRequest.js:77]: this big if sequence followed by a default false return could just be rewritten as 

`const changedState => condition1
 || condition2 
|| condition3 etc `

I'm not sure if thats better really. More concise but maybe not as clear? Up to you

File: src/Service.Host/client/src/components/ChangeRequests/Tabs/BomChangesTab.js [hunk 3]
```diff
@@ -13,11 +14,20 @@ function BomChangesTab({ bomChanges }) {
...
-
         { field: 'bomName', headerName: 'Bom', width: 200 },
-        { field: 'changeState', headerName: 'State', width: 200 },
-        { field: 'phaseInWWYYYY', headerName: 'Phase In Week', width: 200 }
+        {
+            field: 'changeState',
+            headerName: 'State',
+            width: 150,
+            renderCell: params => (
+                <>
```
Comment (lewisrenfrew) [inline src/Service.Host/client/src/components/ChangeRequests/Tabs/BomChangesTab.js:23]: might not need this Fragment? not sure

**Review by lewisrenfrew — APPROVED** [review]
nice

File: src/Service.Host/client/src/components/ChangeRequests/ChangeRequest.js (hunk 2 above)
Comment (andrewfraser73) [reply src/Service.Host/client/src/components/ChangeRequests/ChangeRequest.js:77]: i tried it and it was shorter but denser.  think i prefer the longer more descriptive version

**Pull request merged**